src/Microsoft.AspNetCore.Razor.Language/Legacy/ParserContext.cs

```diff
@@ -49,6 +49,8 @@ public ParserContext(RazorSourceDocument source, RazorParserOptions options)
 
         public bool InTemplateContext { get; set; }
 
+        public bool StartOfLine { get; set; }
```

Comment:
We need this information to handle a few cases properly. Having this as a property here seems cleaner and the information won't be lost when switching between code and markup